### PR TITLE
cleanup code for RecoveyMessages

### PR DESCRIPTION
**Pull request by majianpeng**
@liewegas, log time ago i sent first patch but it don't merge into master(i think it's my fault, see this https://github.com/ceph/ceph/pull/2151).  Send again.

### CODE REVIEW

**Comment by athanatos** [comment]
I had a few other comments, you can re-add needs-qa once you either respond or update the code.

**Comment by majianpeng** [comment]
@athanatos . I reviewed again. I think my second patch did which you comment. Or am i missing somethig? Can you explain in detail? Thanks very much!

**Comment by liewegas** [comment]
@athanatos ping

**Comment by tchaikov** [comment]
@majianpeng could you squash these two change into a single one, so it's more visible that your PR actually addresses the comments from @athanatos already ?

**Comment by majianpeng** [comment]
@tchaikov . updated & please review . Thanks!

**Comment by tchaikov** [comment]
@athanatos i think that @majianpeng has addressed your comments in the latest patch. could you take a look again? thanks. =)

File: src/osd/ECBackend.cc [hunk 1]
```diff
@@ -442,16 +442,20 @@ void ECBackend::dispatch_recovery_messages(RecoveryMessages &m, int priority)
...
-      new SendPushReplies(
-	get_parent(),
-	get_parent()->get_epoch(),
-	replies)));
-  m.t->register_on_applied(
-    new ObjectStore::C_DeleteTransaction(m.t));
-  get_parent()->queue_transaction(m.t);
-  m.t = NULL;
+
+  if (!replies.empty()) {
```
Comment (athanatos) [inline src/osd/ECBackend.cc]: This is not correct, the might be writing out its own copy of the object without replies being populated.  Change this to

if (m.t) {

**Comment by athanatos** [comment]
Caused a bunch of test failures I think due to my above comment.  Ping when fixed or disputed.

File: src/osd/ECBackend.cc (hunk 1 above)
Comment (athanatos) [inline src/osd/ECBackend.cc]: I removed my earlier comments, I was wrong.  However, I'd still like this to be

if (m.t) {

Or, at least, have an

} else {
    assert(!m.t);
}

**Comment by athanatos** [comment]
I'd also like handle_recovery_push to have an assert(m->t) at the top.

**Comment by athanatos** [comment]
This passes tests, I'm just waiting on responses to my comments.

**Comment by tchaikov** [comment]
@athanatos ping ?

**Comment by smithfarm** [comment]
@tchaikov: @athanatos wrote inhttps://github.com/ceph/ceph/pull/3273#issuecomment-103149862 that he is waiting for responses to his comments -- I suppose he means responses from @majianpeng

**Comment by tchaikov** [comment]
@smithfarm, as the response,@majianpeng has addressed sam's comments in his latest commit, that's why i pinged.

**Pull request merged**